include/swift/SIL/SILInstruction.h

```diff
@@ -1222,12 +1222,14 @@ class StringLiteralInst final : public LiteralInst,
 private:
   unsigned Length;
   Encoding TheEncoding;
+  bool NullTerminated;
 
   StringLiteralInst(SILDebugLocation *DebugLoc, StringRef text,
-                    Encoding encoding, SILType ty);
+                    Encoding encoding, bool nullTerminate, SILType ty);
```

Comment:
This `isNullTerminated` flag only needs to be checked when lowering a SLI to an llvm constant, which will only ever happen in exactly one place (`getAddrOfString`).

If `SLI->getValue()` includes the null terminator when it's part of the final value, 5 test cases start failing (I think one dealt with passing down ObjC identifiers to llvm).